### PR TITLE
fixes a bug where `qx.html.Image` breaks JSX

### PR DESCRIPTION
**Pull request by johnspackman**
fixes a bug where `qx.html.Image` is only suitable for `qx.ui.basic.Image` and should not be part of the JSX / VDOM because it breaks the `` tag

### CODE REVIEW

File: source/class/qx/html/Element.js [hunk 1]
```diff
@@ -38,7 +38,8 @@
  *
  * NOTE: Instances of this class must be disposed of after use
  *
- * @require(qx.module.Animation)
+ * NOTE:: This classes used to require `qx.module.Animation` but this brings in a huge
```
Comment (goldim) [inline source/class/qx/html/Element.js:41]:
```suggestion
 * NOTE:: These classes used to require `qx.module.Animation` but this brings in a huge
```

File: source/class/qx/html/Element.js [hunk 2]
```diff
@@ -1678,17 +1678,13 @@ qx.Class.define("qx.html.Element", {
      * @param direct {Boolean?false} Whether the values should be applied
      *    directly (without queuing)
      * @return {qx.html.Element} this object (for chaining support)
-     */
-    setStyles(map, direct) {
+     */ setStyles(map, direct) {
```
Comment (goldim) [inline source/class/qx/html/Element.js:1681]: Broken formatting

File: source/class/qx/html/Node.js [hunk 3]
```diff
@@ -31,7 +31,9 @@
  *
  * NOTE: Instances of this class must be disposed of after use
  *
- * @require(qx.module.Animation)
+ * NOTE:: This classes used to require `qx.module.Animation` but this brings in a huge
```
Comment (goldim) [inline source/class/qx/html/Node.js:34]:
```suggestion
 * NOTE:: These classes used to require `qx.module.Animation` but this brings in a huge
```

**Review by goldim — CHANGES_REQUESTED** [review]
Broken formatting possible somehow to fix?

File: source/class/qx/application/AbstractGui.js [hunk 4]
```diff
@@ -101,9 +102,7 @@ qx.Class.define("qx.application.AbstractGui", {
     // interface method
     close(val) {
       // empty
-    },
-
-    // interface method
+    }, // interface method
```
Comment (derrell) [inline source/class/qx/application/AbstractGui.js:105]: The `// interface method` applies to `terminate`, not to `close`. I think this whitespace was deleted accidentally.

File: source/class/qx/html/Node.js [hunk 5]
```diff
@@ -1261,14 +1263,11 @@ qx.Class.define("qx.html.Node", {
...
-
     /*
     ---------------------------------------------------------------------------
-      PROPERTY SUPPORT
+    PROPERTY SUPPORT
     ---------------------------------------------------------------------------
-    */
-
-    /**
+    */ /**
```
Comment (derrell) [inline source/class/qx/html/Node.js:1270]: Broken formatting

File: source/class/qx/html/Node.js [hunk 6]
```diff
@@ -1284,8 +1283,7 @@ qx.Class.define("qx.html.Node", {
      * @param getter {Function?} function to read from the DOM
      * @param setter {Function?} function to copy to the DOM
      * @param serialize {Function?} function to serialize the value to HTML
-     */
-    registerProperty(key, get, set, serialize) {
+     */ registerProperty(key, get, set, serialize) {
```
Comment (derrell) [inline source/class/qx/html/Node.js:1286]: Broken formatting

File: source/class/qx/html/Node.js [hunk 7]
```diff
@@ -1349,8 +1346,7 @@ qx.Class.define("qx.html.Node", {
      * @param direct {Boolean?false} Whether the value should be applied
      *    directly (without queuing)
      * @return {qx.html.Element} this object (for chaining support)
-     */
-    _setProperty(key, value, direct) {
+     */ _setProperty(key, value, direct) {
```
Comment (derrell) [inline source/class/qx/html/Node.js:1349]: Broken formatting

**Review by derrell — CHANGES_REQUESTED** [review]

File: source/class/qx/html/Element.js (hunk 2 above)
Comment (johnspackman) [reply source/class/qx/html/Element.js:1682]: done

File: source/class/qx/html/Node.js (hunk 6 above)
Comment (johnspackman) [reply source/class/qx/html/Node.js:1287]: done

File: source/class/qx/html/Node.js (hunk 7 above)
Comment (johnspackman) [reply source/class/qx/html/Node.js:1350]: done

**Comment by johnspackman** [comment]
WRT to the formatting issue - prettier keeps breaking things.  I think i've caught them all and put them back, and included turning off the commit hook that was causing it to happen every commit 🙄

**Comment by goldim** [comment]
@johnspackman should this hook be uncommented now after changes?

**Comment by johnspackman** [comment]
> @johnspackman should this hook be uncommented now after changes?

just been trying to make a PR to fix this, but it turns out that it is down to babel mangling the tree before prettier gets involved - and unfortunately, prettier isn't able to fix it.

This means that es6ify should not be used routinely on commits - but prettier should.  I've fixed it, although it won't work on Windows unless there is a bash

**Review by goldim — APPROVED** [review]

**Review by hkollmann — CHANGES_REQUESTED** [review]

File: .github/hooks/pre-commit [hunk 8]
```diff
@@ -7,6 +7,10 @@
 #
 #
 
-qx es6ify --git-pre-commit
+#qx es6ify --git-pre-commit
+for file in $(git diff --cached --name-only --diff-filter=ACMR | grep '\.js') ; do
+  npx prettier --write $file
+done
```
[the remark this answers is not in the document]
Comment (johnspackman) [reply .github/hooks/pre-commit:13]: That would be useful, but I didn't have time and the current implementation breaks code formatting on every commit; I did a quick google, and as I understand it pre-commit hooks don't support windows except as bash scripts - I found an article saying that if you want to run a powershell cmd, you have to write a bash script and get it to run powershell.exe, eg:

```
#!/bin/sh
echo 
exec powershell.exe -ExecutionPolicy RemoteSigned -File '.\pre-commit.ps1'
exit
```

That may be an old article but the previous `pre-commit` had comment lines with `#`, and my Windows says that's not allowed.  Does the hook definitely run on your windows PC?

[the remark this answers is not in the document]
Comment (hkollmann) [reply .github/hooks/pre-commit:13]: Yes the pre-commit with qx es6ify do work. # is'nt a valid comment sign in windows - that's the reason why you get the error.
What's about building a qx prettier command based on the es6ify command? es6ify just do the git calls with the parameter --git-pre-commit.
But this should be another PR.
So i suggest to remove the pre-commit commands completely in this PR and write an issue about creating a prettier command

**Review by hkollmann — APPROVED** [review]

File: .github/hooks/pre-commit (hunk 8 above)
[the remark this answers is not in the document]
Comment (johnspackman) [reply .github/hooks/pre-commit]: done https://github.com/qooxdoo/qooxdoo/issues/10682

**Review by oetiker — APPROVED** [review]

File: source/class/qx/html/Node.js [hunk 9]
```diff
@@ -1264,11 +1267,9 @@ qx.Class.define("qx.html.Node", {
 
     /*
     ---------------------------------------------------------------------------
-      PROPERTY SUPPORT
+    PROPERTY SUPPORT
     ---------------------------------------------------------------------------
-    */
-
-    /**
+    */ /**
```
Comment (derrell) [inline source/class/qx/html/Node.js:1272]: This is showing as still-broken formatting.

**Review by derrell — CHANGES_REQUESTED** [review]

**Review by derrell — APPROVED** [review]

**Review by cboulanger — APPROVED** [review]

**Pull request merged**